helpers.py

```diff
@@ -21,9 +21,78 @@
 import subprocess
 import time
 
-from fabric import Config
-from fabric import Connection
-from paramiko import SSHException
+
+class Result:
+    def __init__(self, stdout, stderr, exited):
+        self.stdout = stdout
+        self.stderr = stderr
+        self.exited = exited
+        self.return_code = exited
+
+
+class Connection:
+    def __init__(self, host, user, port, connect_timeout, key_filename=None):
+        self.host = host
+        self.user = user
+        self.port = port
+        self.connect_timeout = connect_timeout
+        self.key_filename = key_filename
+        self.key = _prepare_key_arg(key_filename)
+
+    def __enter__(self):
+        return self
+
+    def get_ssh_command(self):
+        return f"ssh {self.key} -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o ConnectTimeout={self.connect_timeout} -p {self.port} {self.user}@{self.host}"
+
+    def run(self, command, warn=False, hide=True, echo=False, popen=False):
+        cmd = f"{self.get_ssh_command()} '{command}'"
+
+        # Allows the function to run full ssh and scp commands
+        if not command.startswith("ssh") and not command.startswith("scp"):
+            command = cmd
+
+        logging.debug(command)
+        if echo:
+            print(command)
+
+        if popen:
+            return subprocess.Popen(command)
+        else:
+            try:
+                proc = subprocess.run(
+                    command, check=not warn, capture_output=True, shell=True
+                )
+                returncode = proc.returncode
+
+            except subprocess.CalledProcessError as e:
+                returncode = e.returncode
+                if returncode != 255:
+                    raise
+
+            if returncode == 255:
+                raise ConnectionError(f"Could not connect using command '{command}'")
+
+            stdout = proc.stdout.decode()
+            stderr = proc.stderr.decode()
+
+            if not hide:
+                print(stdout)
+                print(stderr)
+
+            return Result(stdout, stderr, returncode)
+
+    def put(self, file, key_filename=None, local_path=".", remote_path="."):
+        if not key_filename:
+            return put(self, file, self.key_filename, local_path, remote_path)
+        return put(self, key_filename, self.key, local_path, remote_path)
```

Comment:
This is wrong, it should be
```suggestion
        return put(self, file, key_filename, local_path, remote_path)
```